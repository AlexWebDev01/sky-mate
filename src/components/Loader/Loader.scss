@keyframes pulse {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.1);
  }
  100% {
    transform: scale(1);
  }
}

.loader-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100vw;
  height: 100vh;

  .sun-loader {
    display: flex;
    align-self: center;
    justify-self: center;
    margin: auto auto;
    width: 200px;
    height: 200px;
    background-color: var(--orange);
    border-radius: 50%;
    animation: pulse 2s infinite ease-in-out;
  }
}
