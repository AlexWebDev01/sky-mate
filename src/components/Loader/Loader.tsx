import './Loader.scss';

const Loader = () => {
  return (
    <div className='loading-page'>
      <div className='sun-loader'></div>
    </div>
  );
};

export default Loader;
