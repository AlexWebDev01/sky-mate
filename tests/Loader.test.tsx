import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Loader from '@components/Loader/Loader';

jest.mock('@components/Notification/Notification', () => {
  return () => <div data-testid='notification-mock'>Notification</div>;
});

describe('Loader', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(<Loader />);

    const loaderContainer = getByTestId('loader-container');
    expect(loaderContainer).toBeInTheDocument();
    const sunLoader = getByTestId('sun-loader');
    expect(sunLoader).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
