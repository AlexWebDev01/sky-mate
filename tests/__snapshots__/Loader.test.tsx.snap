// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader-container"
    data-testid="loader-container"
  >
    <div
      class="sun-loader"
      data-testid="sun-loader"
    />
  </div>
</DocumentFragment>
`;
